File: src/main/java/com/hrms/hr_payroll_management_system/service/BranchService.java
package com.hrms.hr_payroll_management_system.service;

import com.hrms.hr_payroll_management_system.dto.request.branch.CreateBranchRequest;
import com.hrms.hr_payroll_management_system.dto.request.branch.UpdateBranchRequest;
import com.hrms.hr_payroll_management_system.dto.response.branch.BranchResponse;

import java.util.List;

public interface BranchService {

    BranchResponse create(
            CreateBranchRequest request
    );

    List<BranchResponse> getAll();

    BranchResponse getById(Long id);

    List<BranchResponse> getByCompanyId(
            Long companyId
    );

    BranchResponse update(
            Long id,
            UpdateBranchRequest request
    );

    void delete(Long id);
}

File: frontend/src/services/branch.service.js
import { branchApi } from '../api/branch.api';

export const branchService = {
  async list() {
    const res = await branchApi.getAll();
    return res.data || [];
  },
  async create(payload) {
    const res = await branchApi.create(payload);
    return res.data;
  },
  async update(id, payload) {
    const res = await branchApi.update(id, payload);
    return res.data;
  },
  async remove(id) {
    await branchApi.remove(id);
  },
  async uploadLogo(id, file) {
    const res = await branchApi.uploadLogo(id, file);
    return res.data;
  },
};